modulefiles/module_base.jet

```diff
@@ -1,32 +1,24 @@
+#%Module######################################################################
 ##
-##      nems prerequisites
+##      FV3GFS prerequisites
 ##
 
-proc ModulesHelp {} {
-     puts stderr "The prerequisites for compiling or running FV3 on Jet. "
-}
+# Load hpc-stack
+module use /lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/modulefiles/stack
```

Comment:
This module_base for Jet needs a full overhaul, it does not work on Jet. Please grab this copy from my test clone on Jet:

/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet

I ran successfully with that updated module_base.jet yesterday. Thanks!

Reply:
> This module_base for Jet needs a full overhaul, it does not work on Jet.

The current module_base.jet works. Again, please include error message to justify why it does not work for your test.

Reply:
@lgannoaa See this fcst job log on Jet:
/lfs4/HFIP/hfv3gfs/Kate.Friedman/ptmp/comrot/pr394ff/logs/2021073100/gfsfcst.log.1

The job failed because it didn't know what setpdy was because the prod_util module wasn't loaded because it's wrong in module_base.jet (prod_util/v1.0.18). Upon further inspection of module_base.jet I saw that grib_util was also wrong (not the stack one) and the other modules hadn't been updated to match versions in the stack being used on other platforms. The module_base.jet I pointed you to (which I updated and tested yesterday) was based on module_base.hera with needed updates for Jet (e.g. stack path). Please pull in the module_base.jet version I pointed you to. Thanks!

Reply:
/lfs4/HFIP/hfv3gfs/Lin.Gan/ptmp/com/test2_c192_pr/logs/2020090600/gfsfcst.log line 1174 shown my test successfully run with setpdy.sh executed.
Please redo your test using 97327d. Thanks

Reply:
I updated my clone of your PR branch to the newer hash (97327d) and reran a gfsfcst job with module_base.jet as you have set it and it again failed with the same error. See log here:

/lfs4/HFIP/hfv3gfs/Kate.Friedman/ptmp/comrot/pr394ff/logs/2021073100/gfsfcst.log.0

I then only changed the prod_util module to be the hpc-stack one, reran the gfsfcst again and it's now running successfully:

/lfs4/HFIP/hfv3gfs/Kate.Friedman/ptmp/comrot/pr394ff/logs/2021073100/gfsfcst.log

I am running with a relatively clean account that only loads the stack in my environment. Because of that and the fact that you aren't specifying where that `prod_util/v1.0.18` module can be found the job can't find it at runtime. I'm guessing you have a module use in your Jet .bashrc that helps? I managed to find the `prod_util/v1.0.18` module on Jet and I see that prod_util version is coming from an old copy of NCEPLIBS (/lfs4/HFIP/hfv3gfs/nwprod/NCEPLIBS) which we should be moving away from. So, regardless of all that, you should be using the hpc-stack versions of prod_util and grib_util (and other modules that are available via hpc-stack). You should also be including the other modules that are missing in module_base.jet. So, again, please grab the copy I modified and successfully ran with and add it to your branch. I renamed it before testing your version again so please grab this copy to overwrite your copy of module_base.jet:

/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet_GOOD

Reply:
1. module-setup.sh.inc has module purge on Jet

2. Please explain why your testing module_base.jet is different if you are using my PR branch (97327d) 
diff /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow-pr/modulefiles/module_base.jet
18,19c18
< #module load prod_util/v1.0.18
< module load prod_util/1.2.2
---
> module load prod_util/v1.0.18

**I am unable to reproduce the issue you reported.**
---
[ =0 18:47:04 5792 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> module purge
[ =0 18:47:12 5793 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> which setpdy.sh
/usr/bin/which: no setpdy.sh in ...
module use /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles
module load module_base.jet
which setpdy.sh
**/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/prod_util/1.2.2/bin/setpdy.sh**
---

Reply:
1. module-setup.sh.inc has module purge on Jet
2. Please explain why your testing module_base.jet is different if you are using my PR branch (97327d) 
diff /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow-pr/modulefiles/module_base.jet
18,19c18
< #module load prod_util/v1.0.18
< module load prod_util/1.2.2
---
> module load prod_util/v1.0.18

I am unable to reproduce the issue you reported.
---
[ =0 18:47:04 5792 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> module purge
[ =0 18:47:12 5793 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> which setpdy.sh
/usr/bin/which: no setpdy.sh in ...
module use /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles
module load module_base.jet
which setpdy.sh
/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/prod_util/1.2.2/bin/setpdy.sh
---

Reply:
1. module-setup.sh.inc has module purge on Jet
2. Please explain why your testing module_base.jet is different if you are using my PR branch (97327d) 
diff /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow-pr/modulefiles/module_base.jet
18,19c18
< #module load prod_util/v1.0.18
< module load prod_util/1.2.2
---
> module load prod_util/v1.0.18

I am unable to reproduce the issue you reported.

/lfs4/HFIP/hfv3gfs/Lin.Gan> module purge
[ =0 18:47:12 5793 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> which setpdy.sh
/usr/bin/which: no setpdy.sh in ...
module use /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles
module load module_base.jet
which setpdy.sh
/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/intel-18.0.5.274/prod_util/1.2.2/bin/setpdy.sh

Reply:
> Please explain why your testing module_base.jet is different if you are using my PR branch (97327d)
diff /mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/modulefiles/module_base.jet /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow-pr/modulefiles/module_base.jet
18,19c18
< #module load prod_util/v1.0.18
< module load prod_util/1.2.2

I already explained that, please reread my last comment. You're seeing the one module change I made to get it working (gfsfcst.log) after it failed again with the modules as you have set (gfsfcst.log.0).

What do you have in your .bashrc? Please show what module and PATH settings you have in your .bashrc (e.g. PATH, module use, and module load statements). I am seeing MANY more references to the old NCEPLIBS area (/lfs4/HFIP/hfv3gfs/nwprod/NCEPLIBS) in your gfsfcst.log than either of mine which means your account is getting that path from somewhere.

But, again, this is irrelevant, you should be using hpc-stack modules and have a complete module_base.jet. Your module_base.jet is wrong. Please replace with the modified one I provided.

Reply:
Please keep environment were issue was reported. So other can reproduce the issue as needed.

/lfs4/HFIP/hfv3gfs/Lin.Gan> module purge
[ =0 18:46:38 5788 Lin.Gan@fe8 ]
/lfs4/HFIP/hfv3gfs/Lin.Gan> which setpdy.sh
/usr/bin/which: no setpdy.sh in
module use /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow-pr/modulefiles
/lfs4/HFIP/hfv3gfs/Lin.Gan> module load module_base.jet
/lfs4/HFIP/hfv3gfs/Lin.Gan> which setpdy.sh
/lfs4/HFIP/hfv3gfs/nwprod/NCEPLIBS/src/prod_util_v1.0.18/ush/setpdy.sh
bashrc is not an issue. module prod_util_v1.0.18 loaded provided access to setpdy.sh.

Change made as requested to use hpc-stack.